addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -3830,6 +3830,88 @@ X[]
                     contentAfter: '<p><br></p><p>[]<br></p>',
                 });
             });
+            it('preserve selected table row and insert paragraph in cell when whole table is not selected and enter key is pressed', async () => {
+                // Forward selection
+                await testEditor(BasicEditor, {
+                    contentBefore: unformat(
+                        `<table class="o_selected_table">
+                            <tbody>
+                                <tr><td class="o_selected_td"><p>[<br></p></td><td class="o_selected_td"><p><br></p></td><td class="o_selected_td"><p><br>]</p></td></tr>
+                                <tr><td><p><br></p></td><td><p><br></p></td><td><p><br></p></td></tr>
+                            </tbody>
+                        </table>`,
+                    ),
+                    stepFunction: insertParagraphBreak,
+                    contentAfter: unformat(
+                        `<table>
+                            <tbody>
+                                <tr><td><p><br></p><p>[]<br></p></td><td><p><br></p></td><td><p><br></p></td></tr>
```

Comment:
~~Very strange, when I tested manually I got the extra `p` in the _last_ `td` (the one in which the selection ended, like on GDocs if I'm not mistaken). How come this is different?~~

_Woaw I went and tested on Chrome (I'm on Firefox normally) and the behavior matches what you wrote here. So I got my answer_ :shrug:

UPDATE: I went and tested again and noticed that it works only once per row: ENTER will add a `p` in the last `td`, then select again, press ENTER again, and no extra `p` gets added.

Reply:
Have you solved that part? If so please add tests.

Reply:
How do you produce it? It's working properly for me. As you can see in the [video](https://www.awesomescreenshot.com/video/24664514?key=2b2e625e2b1a267f63a1b9898054fa49)

Reply:
My bad, I just tried and when I thought it wasn't working it was because I was selecting in the opposite direction so the paragraph was added in a different cell and that didn't make the row grow. However, I would have expected the selection to be at the end of the cell after adding the paragraph, and [here](https://www.awesomescreenshot.com/video/24702990?key=84d46cbbc34f7f29450bc1e4c578d65e) it was in the middle of it.

Reply:
This issue occurs because the selection shifts to the 'tr' when multiple rows are selected.
When i try to set the selection at that time, the functionality of selecting table stopped working properly.
so, to address this problem, we need to change how table is being selected. Would you like me to proceed with it in this task?

Reply:
It depends on what you mean with "change how table is being selected"... The feature should behave as expected so, yes, but that shouldn't mean rewriting the whole feature. Try to make this work in a separate commit so we can see at what cost the fix comes.